Xplat/src/main/java/vazkii/botania/common/item/GrassSeedsItem.java

```diff
@@ -35,17 +35,28 @@ public class GrassSeedsItem extends Item implements FloatingFlowerVariant {
 	 * active in that dimension.
 	 */
 	private static final Map<ResourceKey<Level>, Set<BlockSwapper>> blockSwappers = new HashMap<>();
-	private static final Map<IslandType, float[]> COLORS = ImmutableMap.<IslandType, float[]>builder()
-			.put(IslandType.GRASS, new float[] { 0F, 0.4F, 0F })
-			.put(IslandType.PODZOL, new float[] { 0.5F, 0.37F, 0F })
-			.put(IslandType.MYCEL, new float[] { 0.27F, 0F, 0.33F })
-			.put(IslandType.DRY, new float[] { 0.4F, 0.5F, 0.05F })
-			.put(IslandType.GOLDEN, new float[] { 0.75F, 0.7F, 0F })
-			.put(IslandType.VIVID, new float[] { 0F, 0.5F, 0.1F })
-			.put(IslandType.SCORCHED, new float[] { 0.75F, 0F, 0F })
-			.put(IslandType.INFUSED, new float[] { 0F, 0.55F, 0.55F })
-			.put(IslandType.MUTATED, new float[] { 0.4F, 0.1F, 0.4F })
+	private static final Map<IslandType, Integer> IDS = ImmutableMap.<IslandType, Integer>builder()
+			.put(IslandType.GRASS, 0)
+			.put(IslandType.PODZOL, 1)
+			.put(IslandType.MYCEL, 2)
+			.put(IslandType.DRY, 3)
+			.put(IslandType.GOLDEN, 4)
+			.put(IslandType.VIVID, 5)
+			.put(IslandType.SCORCHED, 6)
+			.put(IslandType.INFUSED, 7)
+			.put(IslandType.MUTATED, 8)
```

Comment:
We shouldn't have an ID system here, that's one more thing that could go wrong especially with addon supplied island types.
Instead, I'd do the following:

1. Keep the existing map here, but replace the floats with their packed integer hex-color equivalents
2. Grab the color from this map serverside and send it in the packet in the DispenserBehavior
3. The client just uses the color given in the packet.

Reply:
I realize what I suggest still isn't open to addons, but it at least avoids yet another ID system